core/native-bridge.ts

```diff
@@ -513,6 +513,22 @@ const initBridge = (w: any): void => {
             method.toLocaleUpperCase() === 'OPTIONS' ||
             method.toLocaleUpperCase() === 'TRACE'
           ) {
+            // a workaround for following android webview issue:
+            // https://issues.chromium.org/issues/40450316
+            // Sets the user-agent header to a custom value so that its not stripped
+            // on its way to the native layer
+            if (platform === 'android' && options) {
+              if (headers.has('User-Agent')) {
+                headers.set('X-Cap-User-Agent', headers.get('User-Agent') as string);
+              }
+
+              if (headers.has('user-agent')) {
+                headers.set('X-Cap-User-Agent', headers.get('user-agent') as string);
+              }
+
+              options.headers = headers;
+            }
```

Comment:
This can be simplified a little. Also `x-cap-user-agent` should be lowercased, despite is being lowercased automatically, but for consistency.
```suggestion
            if (platform === 'android' && options?.headers) {
              const userAgent = headers.get('User-Agent') || headers.get('user-agent');
              if (userAgent) {
                headers.set('x-cap-user-agent', userAgent);
                options.headers = headers;
              }
            }
```